src/abstractdataframe/abstractdataframe.jl

```diff
@@ -405,31 +412,109 @@ describe(df)
+function StatsBase.describe(io, df::AbstractDataFrame; colstats = [:mean, :min, :median, :max, :Nmissing, :datatype])
+    # Check that people don't specify the wrong fields. 
+    allowed_fields = [:mean, :sd, :min, :q25, :median, :q75, :max, :datatype, :Nunique, :Nmissing]
+    for i in colstats 
+        if !contains(==, allowed_fields, i) 
+            error("""
+                Not an allowed field. Allowed fields are:
+                :mean,
+                :min,
+                :q25,
+                :median,
+                :q75,
+                :max,
+                :datatype,
+                :Nmissing,
+                :Nunique""") 
+        end
     end
-end
-
-function StatsBase.describe(io::IO, X::AbstractVector{Union{T, Missing}}) where T
-    missingcount = count(ismissing, X)
-    pmissing = 100 * missingcount/length(X)
-    if pmissing != 100 && T <: Real
-        show(io, StatsBase.summarystats(collect(skipmissing(X))))
-    else
-        println(io, "Summary Stats:")
+    # Define 4 functions for getting summary statistics 
+    # use a dict because we dont know which measures the user wants
+    function get_stats(col::AbstractArray{T} where T <: Real)
+        stats = summarystats(col)
+        stats_dict = Dict(
+            :mean => stats.mean,
+            :sd => sqrt(var(col)),
+            :min => stats.min ,
+            :q25 => stats.q25,
+            :median => stats.median,
+            :q75 => stats.q75,
+            :max => stats.max,
+            :datatype=> eltype(col),
+            :Nmissing => nothing,
+            :Nunique => nothing
+        )
+    end
+    
+    function get_stats(col::AbstractArray{Union{T, Missing}} where T <: Real)
+        stats = summarystats(collect(skipmissing(col)))
+        stats_dict = Dict(
+            :mean => stats.mean,
+            :sd => sqrt(var(collect(skipmissing(col)))),
+            :min => stats.min ,
+            :q25 => stats.q25,
+            :median => stats.median,
+            :q75 => stats.q75,
+            :max => stats.max,
+            :datatype=> Missings.T(eltype(col)),
+            :Nmissing => count(ismissing(col)),
+            :Nunique => nothing
+        )
     end
-    println(io, "Length:         $(length(X))")
-    println(io, "Type:           $(eltype(X))")
-    !(T <: Real) && println(io, "Number Unique:  $(length(unique(X)))")
-    println(io, "Number Missing: $(missingcount)")
-    @printf(io, "%% Missing:      %.6f\n", pmissing)
-    return
+    
+    function get_stats(col)
+        stats_dict = Dict(
+            :mean => nothing,
+            :sd => nothing,
+            :min => nothing,
+            :q25 => nothing,
+            :median => nothing,
+            :q75 => nothing,
+            :max => nothing,
+            :datatype=> eltype(col),
+            :Nmissing => nothing,
+            :Nunique => length(unique(col))
+        )   
+    end
+    
+    function get_stats(col:: AbstractArray{Union{T, Missing}} where T <: Any)
```

Comment:
Just `col::AbstractArray`, since `Union{Any, T} == Any`.

Reply:
Could you please clarify this? How else can we dispatch on the presence of missing values? Since we want it to return `nothing` sometimes and the number of missing values other times.